### PR TITLE
Implement S5::drawScenarioPreviewImage

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Implements the `S5::drawScenarioPreviewImage` function. During implementation, I ran this implementation and the vanilla function side-by-side to ensure a correct result. The code has since been cleaned up, and is now ready to be merged.

### CODE REVIEW

File: src/OpenLoco/src/S5/S5.cpp [hunk 1]
```diff
@@ -109,7 +113,71 @@ namespace OpenLoco::S5
...
+        auto& options = S5::getOptions();
+        const auto kPreviewSize = sizeof(options.preview[0]);
+        const auto kMapSkipFactor = kMapRows / kPreviewSize;
+
+        for (auto y = 0U; y < kPreviewSize; y++)
+        {
+            for (auto x = 0U; x < kPreviewSize; x++)
+            {
+                uint8_t colour = 0;
+                auto pos = TilePos2(kMapColumns - (x + 1) * kMapSkipFactor + 1, y * kMapSkipFactor + 1);
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/S5/S5.cpp:128]: This is some deep nesting can we split it off into a function

Comment (AaronVanGeffen) [reply src/OpenLoco/src/S5/S5.cpp:128]: Could move the tile colour stuff into a function of its own, I suppose. Something like `getPreviewColourByTilePos`?

**Review by LeftofZen — APPROVED** [review]
It was extremely satisfying to line screenshots of the map previews up from master and your branch and apply a difference filter in photoshop and see every single pixel match exactly. Nice work!

**Review by LeftofZen — DISMISSED** [review]
-wrong comment-

**Comment by AaronVanGeffen** [comment]
Hmm, both these traces are unrelated to the actual river carving. I suspect things are going sour further down the generation process.  I'll have to find a way to reliably reproduce the issue before I can debug it though.

**Review by LeftofZen — APPROVED** [review]

**Pull request merged**